Mathlib/GroupTheory/Coprod/Basic.lean

```diff
@@ -573,9 +573,9 @@ theorem mk_of_inv_mul : ∀ x : G ⊕ H, mk (of (x.map Inv.inv Inv.inv)) * mk (o
 theorem con_inv_mul_cancel (x : FreeMonoid (G ⊕ H)) :
     coprodCon G H (ofList (x.toList.map (Sum.map Inv.inv Inv.inv)).reverse * x) 1 := by
   rw [← mk_eq_mk, map_mul, map_one]
-  induction x with
-  | h0 => simp [map_one mk] -- TODO: fails without `[map_one mk]`
-  | ih x xs ihx =>
+  induction x using FreeMonoid.inductionOn' with
+  | one => simp [map_one mk] -- TODO: fails without `[map_one mk]`
```

Comment:
Is the TODO still relevant? ie does this work?
```suggestion
  | one => simp
```

Reply:
looks like that TODO is no longer relevant - it works with just "simp"

Reply:
yup, that's a nice thing!